events/importer/kulke.py

```diff
@@ -851,88 +871,84 @@ def simple(field):
-                        superevent_aggregates.values_list("id", flat=True), group
-                    )
+    @transaction.atomic
+    def _save_recurring_superevent(self, recurring_group):
+        kulke_ids = set(make_kulke_id(event) for event in recurring_group)
+        superevent_aggregates = EventAggregate.objects.filter(
+            members__event__id__in=kulke_ids
+        ).distinct()
+        n_super_events = superevent_aggregates.count()
+
+        if n_super_events > 1:
+            logger.error(
+                "Error: the superevent has an ambiguous aggregate group.\nAggregate ids: {}, group: {}".format(
+                    superevent_aggregates.values_list("id", flat=True), recurring_group
                 )
-                continue
+            )
+            return False
 
-            events = Event.objects.filter(id__in=kulke_ids)
-            if events.count() < 2:
-                continue
+        events = list(Event.objects.filter(id__in=kulke_ids))
+        if len(events) != len(recurring_group):
+            logger.warning(
+                "Not all events referenced in the group were found in the database. Group: %s - Events: %s",
+                recurring_group,
+                set(e.id for e in events),
+            )
 
-            aggregate = None
-            if cnt == 0:
-                if len(group) == 1:
-                    # Do not create aggregates of only one.
-                    continue
-                aggregate = EventAggregate()
-                aggregate.save()
-                super_event = Event(
-                    publisher=self.organization,
-                    super_event_type=Event.SuperEventType.RECURRING,
-                    data_source=DataSource.objects.get(pk="kulke"),  # TODO
-                    id="linkedevents:agg-{}".format(aggregate.id),
-                )
-                super_event.save()
-                aggregate.super_event = super_event
-                aggregate.save()
+        aggregate = None
+        if n_super_events == 0:
+            # Don't create a super event if there is only one event in the database
+            if len(events) < 2:
+                return False
+
+            aggregate = EventAggregate()
+            aggregate.save()
+            super_event = Event(
+                publisher=self.organization,
+                super_event_type=Event.SuperEventType.RECURRING,
+                data_source=self.data_source,
+                id="linkedevents:agg-{}".format(aggregate.id),
+            )
+            self._update_super_event(super_event, events)
+            super_event.save()
+            aggregate.super_event = super_event
+            aggregate.save()
+            event_aggregates = [
+                EventAggregateMember(event=event, event_aggregate=aggregate)
+                for event in events
+            ]
+            EventAggregateMember.objects.bulk_create(event_aggregates)
+        elif n_super_events == 1:
```

Comment:
Here `n_super_events` must be `1` because we have already checked that it cannot be `>1` or `0`.
```suggestion
        else:
```

Reply:
Fixed